core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1927,7 +1930,8 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     for {
       googleProjectId <- traceWithParent("getGoogleProjectFromBuffer", span)(_ => resourceBufferService.getGoogleProjectFromBuffer(projectPoolType, workspaceId))
       _ <- traceWithParent("updateGoogleProjectBillingAccount", span)(_ => gcsDAO.updateGoogleProjectBillingAccount(googleProjectId, Option(billingAccount)))
-      googleProjectNumber <- traceWithParent("getProjectNumberFromGoogle", span)(_ => getGoogleProjectNumber(googleProjectId))
+      googleProjectLabels = Map("workspaceNamespace" -> workspaceName.namespace, "workspaceName" -> workspaceName.name, "workspaceId" -> workspaceId)
+      googleProjectNumber <- traceWithParent("setUpProjectInCloudResourceManagerAndGetGoogleProjectNumber", span)(_ => setUpProjectInCloudResourceManagerAndGetGoogleProjectNumber(googleProjectId, googleProjectLabels))
```

Comment:
This feels a little clunky to me now.  Perhaps we try something like:
```suggestion
      googleProject <- traceWithParent("setUpProjectInCloudResourceManager", span)(_ => setUpProjectInCloudResourceManager(googleProjectId, googleProjectLabels))
      googleProjectNumber = gcsDAO.getGoogleProjectNumberFromOption(googleProjectId, Option(GoogleProjectNumber(googleProject.getProjectNumber.toString)))
```
Of course `setUpProjectInCloudResourceManager` would need to be updated to return the project, but that looks trivial.

My preference is to try not to have a single function doing too much.  The important thing this function is doing is the project setup.  The fact that it returns a Project or a Project Number is secondary, so l think this code becomes more maintainable if the steps are separated.